### PR TITLE
Cleanup: remove potential goroutine leakages in taskrun

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [ ] Release notes block below has been filled in
(if there are no user facing changes, use release note "NONE")

# Release Notes
NONE

### CODE REVIEW

**Comment by kerthcet** [comment]
> /ok-to-test
> 
> Thanks @kerthcet! would you mind adding a commit message body please?

Absolutely, done.

**Comment by abayer** [comment]
/lgtm

Good catch!

**Comment by kerthcet** [comment]
@abayer PTAL, fixed the tests error, thanks.

**Comment by kerthcet** [comment]
Well, unit tests failed for some unknown reason, details here https://tekton-releases.appspot.com/build/tekton-prow/pr-logs/pull/tektoncd_pipeline/4936/pull-tekton-pipeline-unit-tests/1532906825693794306/, but I can't reproduce this, all tests passed in my local env, unhold then.
/unhold

File: pkg/taskrunmetrics/metrics_test.go [hunk 1]
```diff
@@ -64,16 +64,18 @@ func TestUninitializedMetrics(t *testing.T) {
 		Status: corev1.ConditionUnknown,
 	}
 
-	if err := metrics.DurationAndCount(&v1beta1.TaskRun{}, beforeCondition); err == nil {
+	ctx, cancel := context.WithCancel(context.Background())
```
Comment (lbernick) [inline pkg/taskrunmetrics/metrics_test.go:67]: just curious, what's the need for the cancellation function here?

**Comment by lbernick** [comment]
> Well, unit tests failed for some unknown reason, details here https://tekton-releases.appspot.com/build/tekton-prow/pr-logs/pull/tektoncd_pipeline/4936/pull-tekton-pipeline-unit-tests/1532906825693794306/, but I can't reproduce this

Unfortunately our tests are quite flaky right now :( This error looks a lot like a flake to me so OK to rerun

File: pkg/taskrunmetrics/metrics_test.go (hunk 1 above)
Comment (kerthcet) [reply pkg/taskrunmetrics/metrics_test.go:67]: Make sure to teardown the resources.

**Comment by lbernick** [comment]
@abayer I don't think anything has really changed since you LGTMed so I'll go ahead and add that label-- lmk if any concerns

/lgtm

**Comment by kerthcet** [comment]
It seems like I have no privilege to remove the `do-not-merge/hold` label, need some help @lbernick , thanks a lot.

**Comment by lbernick** [comment]
oh oops sorry-- I don't think you need permissions, it's just a different command

/hold cancel

**Pull request merged**